lib/src/util/io.dart

```diff
@@ -130,8 +130,12 @@ void warn(String message, {bool color}) {
 /// applications that don't print out the bound port.
 Future<T> getUnusedPort<T>(FutureOr<T> tryPort(int port)) async {
   T value;
+  var supportsIPv6 = false;
+  if (await _supportsIPv6()) {
+    supportsIPv6 = true;
+  }
```

Comment:
I don't understand that reasoning. If you recorded whether IPv6 binding worked, you'd only call `bind()` twice the first time through the loop.